exporters/cinder.go

```diff
@@ -308,7 +312,52 @@ func ListVolumeLimits(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metr
 		if err != nil {
 			return err
 		}
-
+		jsonLimits := quotasets.GetUsage(exporter.Client, p.ID).PrettyPrintJSON()
+		var jsonParse map[string]interface{}
+		_ = json.Unmarshal([]byte(jsonLimits), &jsonParse)
+		quota := jsonParse["quota_set"].(map[string]interface{})
+		for i := range quota {
+			words := strings.Split(i, "_")
+			wordsLen := len(words)
+			if wordsLen > 1 {
+				if words[0] == "gigabytes" {
+					if wordsLen == 2 {
+						disksQuota := quota[i].(map[string]interface{})
+						inUseGb := disksQuota["in_use"].(float64)
+						volType := words[1]
+						ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_type_used_gb"].Metric,
+							prometheus.GaugeValue, inUseGb, p.Name, p.ID, volType)
+					} else if wordsLen > 2 {
+						jsonHead := strings.Join(words, "_")
+						disksQuota := quota[jsonHead].(map[string]interface{})
+						inUseGb := disksQuota["in_use"].(float64)
```

Comment:
casting may fail and break the exporting

Reply:
fixed in 085a1ae20af7403bcf732e4dc0d2089c129747f3